lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -865,3 +865,79 @@ def test_course_catalog_access_num_queries(self, user_attr_name, action, course_
         course_overview = CourseOverview.get_from_id(course.id)
         with self.assertNumQueries(num_queries, table_blacklist=QUERY_COUNT_TABLE_BLACKLIST):
             bool(access.has_access(user, action, course_overview, course_key=course.id))
+
+
+@ddt.ddt
+class CCXTestCase(SharedModuleStoreTestCase):
+    """
+    Tests for ccx course visibility
+    """
+    MODULESTORE = TEST_DATA_SPLIT_MODULESTORE
+
+    def setUp(self):
+        """
+        Set up tests
+        """
+        super(CCXTestCase, self).setUp()
+
+        course = CourseFactory.create(enable_ccx=True)
+        CourseOverview.load_from_module_store(course.id)
+        self.course = course
+
+        student = UserFactory.create()
+        staff = AdminFactory.create()
+        coach = UserFactory.create()
+
+        ccx = CustomCourseForEdX(
+            course_id=course.id,
+            coach=coach,
+            display_name="Test CCX"
+        )
+        ccx.save()
+        self.ccx_overview = CourseOverview.get_from_id(ccx.locator)
+
+        self.users = {
+            'coach': coach,
+            'student': student,
+            'staff': staff,
+        }
+
+    @ddt.data(
+        ('student', 'see_in_catalog', True),
+        ('student', 'see_exists', True),
+    )
+    @ddt.unpack
+    def test_base_course_visibility(self, username, permission, has_permission):
+        """
+        Test base course visibility is as expected
+        """
+        user = self.users[username]
+        self.assertEqual(bool(access.has_access(user, permission, self.course)), has_permission)
+
+    @ddt.data(
+        ('student', 'see_in_catalog', False),
+        ('student', 'see_exists', False),
+        ('student', 'enroll', False),
+        # load permission is required because other areas in the code need it.
```

Comment:
Why do other areas need it?

Reply:
To be honest, I'm not sure at this stage. The 'load' permission is used for a lot of different things, and I seem to remember that modifying it caused issues. From memory it was something relating to needing to access the about page to enroll. Sorry I can't exactly remember and can't trivially find it again. Will investigate more.